### PR TITLE
Updated Node.js dgram.send() signature with recent changes in Node.js API

### PR DESCRIPTION
**Pull request by nebrius**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/dist/latest-v8.x/docs/api/dgram.html#dgram_socket_send_msg_offset_length_port_address_callback (see history section)
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/node/v8/index.d.ts [hunk 1]
```diff
@@ -2850,8 +2850,8 @@ declare module "dgram" {
     export function createSocket(options: SocketOptions, callback?: (msg: Buffer, rinfo: RemoteInfo) => void): Socket;
 
     export class Socket extends events.EventEmitter {
-        send(msg: Buffer | String | any[], port: number, address: string, callback?: (error: Error | null, bytes: number) => void): void;
-        send(msg: Buffer | String | any[], offset: number, length: number, port: number, address: string, callback?: (error: Error | null, bytes: number) => void): void;
+        send(msg: Buffer | String | Uint8Array | any[], port: number, address: string, callback?: (error: Error | null, bytes: number) => void): void;
```
Comment (Flarna) [inline types/node/v8/index.d.ts]: I think it should be `string` instead `String` here.
According to docs address is optional.

File: types/node/v8/index.d.ts [hunk 2]
```diff
@@ -2850,8 +2850,8 @@ declare module "dgram" {
...
+        send(msg: Buffer | String | Uint8Array | any[], offset: number, length: number, port: number, address: string, callback?: (error: Error | null, bytes: number) => void): void;
```
Comment (Flarna) [inline types/node/v8/index.d.ts]: docs tell `If msg is an array, offset and length must not be specified.` therefore I think `| any[]` should be removed here.

**Review by Flarna — CHANGES_REQUESTED** [review]
Please apply the change also to Node 10.

**Comment by nebrius** [comment]
> Please apply the change also to Node 10.

Whoops, my local branch was just barely out of date enough to not have the v10 docs yet, will fix. Everything else makes sense too.

**Comment by nebrius** [comment]
Looks like I spoke too soon, there doesn't appear to be v10 docs yet: https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/node

**Comment by Flarna** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/index.d.ts is latest which is Node 10 since ~2 weeks.

**Comment by nebrius** [comment]
Ohhhh, thanks, I didn't pick up that the root index.d.ts was the most recent version.

**Comment by nebrius** [comment]
All comments should be addressed. Note: I also marked the `address` parameter optional to conform to the docs.

**Comment by nebrius** [comment]
Travis failure seems to be spurious?

screen shot 2018-05-10 at 14 28 42

**Review by Flarna — APPROVED** [review]

**Comment by Flarna** [comment]
I agree that CI failure is unrelated. I have seen this recently in #25597 where retrigger CI (by closing and reopening) "solved" this - even log showed the following:
```
redux-form/v6 OK
The 'no-void-expression' rule threw an error in '/home/travis/build/DefinitelyTyped/DefinitelyTyped/types/redux-little-router/redux-little-router-tests.tsx':
RangeError: Maximum call stack size exceeded
    at isPossiblyReferencedInConditionalType (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/typescript/lib/typescript.js:34309:17)
    at /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/typescript/lib/typescript.js:34328:146
    at Object.filter (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/typescript/lib/typescript.js:1969:31)
    at getTypeFromConditionalTypeNode (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/typescript/lib/typescript.js:34328:92)
    at getTypeFromTypeNode (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/typescript/lib/typescript.js:34695:28)
    at getTemplateTypeFromMappedType (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/typescript/lib/typescript.js:32020:52)
    at substituteIndexedMappedType (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/typescript/lib/typescript.js:34165:36)
    at getSimplifiedIndexedAccessType (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/typescript/lib/typescript.js:34157:28)
    at getSimplifiedType (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/typescript/lib/typescript.js:34110:63)
    at isRelatedTo (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/typescript/lib/typescript.js:35594:30)
```
@andy-ms Any hint regarding this spurious failures?

**Comment by andy-ms** [comment]
I'll look into that soon -- need to improve the error message from "something went wrong"

**Pull request merged**